### ISSUE TITLE
Prototype/Design Update - Hand Off of CFR from IAO to Ministry (Change of State)

### ISSUE THREAD

**Issue by JHarrietha-AOT** [bcgov/foi-flow#443]
Description: 
First draft of design for Call for Records Hand off process from IAO to Ministries. Based off of the user flow created as a product of the UX Research Sessions with Ministries.

#### DOD
- [x] Designing Process for Change of State
- [x] How this will be passed off from IAO to Ministries
- [x] Change of state text box that prompts users to add any additional notes with text box (Add LAN link for just general hand off know
- [x] Ability to back track to IAO if further clarifications are required and CFR can not begin
- [x] Add option for (unlimited) number of 'watchers' which can assign themselves to others to watch a file [ie, IAO would be a watcher of requests in Call for Records]
- [x] Removing Open Button as Bottom screen - user will need to change state in side bar and have pop up window prompting them to select or see who is assigned. and they can either move forward with new assignment or cancel
- [x] Updating Side Nav to display both LDD and Records Due back to IAO

**Comment by JHarrietha-AOT** [comment]
@lmullane @mpilchar @arielleandrews 

Below is a link to the updated design for the changing of state. As a note the above ticket does not include the activity log or the content of the Call for Records Page and Assigning and Tracking Divisions; these will come in ticket #363

The focus of these design changes in the changes of state, the modal and the functionality of moving between states. Please review and let me know what changes and suggestions you each have when you have a chance. 

You can click on IAO Analyst and follow through the change of state from 
Intake in Progress > Open
Open > Call for Records
Call for Records < Open
If you click on Ministry sign in you will be able to see the Ministry Dashboard. 

https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/657932450

**Comment by lmullane** [comment]
Thanks. @JHarrietha-AOT 

I'll take a look and post any comments/questions here.

**Comment by lmullane** [comment]
@JHarrietha-AOT, looking at the designs I see we only have 1 assigned too (i.e either Ministry or IAO team/analyst).

I was assuming we would have 2 assignments: 1 to ministry and 1 to IAO analyst. In this scenario, an analyst could change who at IAO was assigned, but the ministry assignment could remain unchanged.

In addition by having 2 assignees, both queues would be populated with the request.

How were you thinking it would work with only 1 assignee? 

cc @mpilchar, @arielleandrews, @abin-aot

**Comment by JHarrietha-AOT** [comment]
@lmullane 

I was thinking it would function as the core person doing the work on that given request would be the assigned. So when in Call for Records the Ministry would be assigned and IAO would be able to add themselves (Or could be automatically added) under the 'Watching Request' dropdown where multiple users could be selected. 

Happy to update the design to have a 'IAO Assigned to' 'Ministry Assigned To' and then 'Watching Request' So it is more explicit.

Let me know where the preference lies, that would be a small update I could do quickly.

**Comment by mpilchar** [comment]
Nicely done Jackie! Weighing in on the discussion - When covering the "assignee" and "watcher" previously, having clear visibility into who was driving a request at any given time was valued highly, with the watcher functionality giving transparency to others wanting to stay informed on that request. @lmullane it seems like the current design achieves the goal of clear visibility - do you foresee it creating confusion? Will the ministries know who in IAO is their contact for the request with the current design?

**Comment by lmullane** [comment]
I figured you were thinking of the watcher feature in that way.

I do wonder if the IAO team lead may want to have a clear assignee on their team? There may be many watchers. Let's hold until @arielleandrews chimes in.

**Comment by JHarrietha-AOT** [comment]
Sounds good, we can wait till she is back.

**Comment by JHarrietha-AOT** [comment]
Only IAO Flex team should have the ability to return a request back to the OPEN state. Ministries should not be able to do this.

Changes of State: 
Add window for 'Ministry Assigned To:' "IAO Assigned To:'
Comments should be removed for MVP
Ministry should not be changeable - remove dropdown option
Watcher DropDown - everyone should have functionality to be able to see and assign others to be a watcher - on CFR Modal it just displays who the watchers are, 
Queue Needs to have updated Toggle (Requests Assigned, Request Watching, All Requests) - Team Lead should sign in and have default queue of all, vs Analyst should sign in and have default queue of Assigned Requests.
Ability to select if user wants to be notified as a watcher - ? - What do you want to know as a watcher

**Comment by JHarrietha-AOT** [comment]
@mpilchar @arielleandrews  @lmullane 

I have updated the design of the modal as discussed this morning based off of the notes I took above. I have moved the Watcher dropdown to the request description page as well. It is currently based off of a multi select dropdown in Material UI. I am going to revisit in hopes of doing something closer to GitHub watcher. I will include this in #363 as we have removed the design of the activity log. 

Let me know if I missed anything from our discussion this morning or have any further changes
